app/src/main/java/org/oppia/android/app/player/state/StatePlayerRecyclerViewAssembler.kt

```diff
@@ -1306,6 +1385,7 @@ class StatePlayerRecyclerViewAssembler private constructor(
     val replaySupport: Boolean = false,
     val returnToTopicNavigation: Boolean = false,
     val showCelebrationOnCorrectAnswer: Boolean = false,
+    val showCelebrationAtEndOfExplorationSession: Boolean = false,
```

Comment:
We shouldn't specialize the assembler specifically for explorations or questions. Might: ``showCelebrationOnSessionCompletion`` be clearer? We probably won't enable this for questions since we'd need a way to specifically differentiate successful completions, but this one is still clear: it's always showing a celebration at the end. WDYT?

Ditto everywhere else here where we're now referencing to an exploration.

Reply:
My intention for specifying explorations was because of upcoming UI changes for the end of practice sessions that affects whether it will show confetti, making it easier to add a different confetti animation for practice sessions in the future. Also, because this assembler handles both the exploration player and the question player it felt better to clarify that this feature was meant for explorations. (In retrospect, both these reasonings aren't totally accurate as it would still be the same feature, just with a different implementation)

Looking at the flow of data from presenter => assembler, I agree its illogical to be having the question player setting a feature for end of explorations -- will change the name.

Reply:
Done

Reply:
I suspect in that case it may be a matter of customizing the end-of-session experience rather than creating a separate feature. Another way to think about this: the assembler doesn't know what an exploration or question session is, nor should it. By specializing features for the usage of the assembler we're effectively "leaking state" into the assembler implementation which conflates its own responsibilities.

Reply:
This all makes sense and I agree with the idea of customizing the end-of-session experience rather than being a different feature.